conf/application.conf

```diff
@@ -227,202 +227,501 @@ governmentSpend {
   2014 {
 
     percentages {
-      Welfare = 24.52
+      1 {
```

Comment:
Is this the ordering that used to be in the frontend? So we've moved it into the backend? If this is how we're doing it, do we even need numbers in here? Could we not just list the categories in order in the file without number + generate the numbers if needed when read in?

Reply:
Yes, Its the ordering that used to be in the frontend. And the reason we moved it backend is because there should be only one place to manage it, Else there might be discrepancy.
2 Approaches that you spoke about :
1) Maintain the separate list for the categories: If you maintain the separate list then there are still 2 places where you maintain the categories. One for the percentage and other for the order
2) If you just list the categories in order without the numbers: Its just becomes map and cannot be read as listMap or list. And map does not maintain the order. so to read it from the config you need explicit ordering somewhere.

Reply:
You could possibly read the items in as a list initially then parse each item as a key value pair. It might be worth trying that as it will result in a simpler config format. Also, as there's some processing involved here could we have a unit test for the governmentSpend method in ApplicationConfigSpec?

Reply:
I tried to make it list and read it as. a list without the index. But it did not seem to be possible with all the syntaxes .map comes into picture and as soon as that happens the ordering is broken. I will write the unit test for the governmentSpend

Reply:
Yeah, the problem is the config is based around a Set which is unordered. Could put ordering in as a separate field, e.g. ordering=Health,... But it's probably not worth the trouble.

Reply:
Added the Spec